src/main/java/de/tum/cit/aet/artemis/core/service/connectors/SAML2Service.java

```diff
@@ -135,6 +138,29 @@ public Authentication handleAuthentication(final Authentication originalAuth, fi
                 }
             }
         }
+        else if (saml2syncUserData.isPresent() && Boolean.TRUE.equals(saml2syncUserData.get())) {
+            log.debug("SAML2 sync user data enabled and will be performed for user {}", user.get().getLogin());
+            // We assume that only the name of the user might change
+            String newFirstName = substituteAttributes(properties.getFirstNamePattern(), principal);
+            String newLastName = substituteAttributes(properties.getLastNamePattern(), principal);
+            String oldFirstName = user.get().getFirstName();
+            String oldLastName = user.get().getLastName();
+
+            boolean changed = false;
+            if (!oldFirstName.equals(newFirstName)) {
+                user.get().setFirstName(newFirstName);
+                changed = true;
+            }
+            if (!oldLastName.equals(newLastName)) {
+                user.get().setLastName(newLastName);
+                changed = true;
+            }
+
+            if (changed) {
+                log.info("SAML2 user's name changed ... before: {} {}, after: {} {}", oldFirstName, oldLastName, newFirstName, newLastName);
+                userRepository.save(user.get());
+            }
```

Comment:
nitpick: could probably be moved into its dedicated method (did not check which additional parameters would be needed)

```suggestion
            user = Optional.of(syncUserInformation(user.get()));
```

```java
private User syncUserInformation(final User user /* ... */) {
    // ...
    if (changed) {
        log.info("SAML2 user's name changed ... before: {} {}, after: {} {}", oldFirstName, oldLastName, newFirstName, newLastName);
         return userRepository.save(user);
    } else {
        return user;
    }
}
```

Reply:
True .. done :)